### PR TITLE
FUSETOOLS2-159: Add UI Tests for auto-completion inside Generic Text …

### PR DESCRIPTION
**Pull request by pospisilf**
…Editor

# Pull Request informations

## Rebase & Merge default requirements

1. Green build for main branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **main** branch build

### CODE REVIEW

**Comment by pospisilf** [comment]
https://github.com/camel-tooling/camel-lsp-client-eclipse/pull/158 must be merged first

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/SetDefaultEditor.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.eclipse.reddeer.swt.impl.button.OkButton;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Sets default editor for file type Eclipse preference.
+ * @author fpospisi
+ */
+public class SetDefaultEditor extends PreferencePage {
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/SetDefaultEditor.java:17]: this class is not used, could be deleted?

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/tests/com.github.cameltooling.lsp.ui.tests/src/com/github/cameltooling/lsp/ui/tests/suite/SmokeTests.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/tests/com.github.cameltooling.lsp.ui.tests/src/com/github/cameltooling/lsp/ui/tests/suite/SmokeTests.java:1]: could you please change license to ASF as it is in other files? thank you

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/DefaultEditorPreferencePage.java [hunk 3]
```diff
@@ -0,0 +1,414 @@
+/**
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/DefaultEditorPreferencePage.java:1]: could you please add missing ASF license? thanks

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/FileTypeDialog.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/FileTypeDialog.java:1]: could you please add missing ASF license? thanks

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/utils/CreateNewEmptyFile.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package com.github.cameltooling.lsp.reddeer.utils;
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/utils/CreateNewEmptyFile.java:1]: could you please add missing ASF license? thanks

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/utils/JavaProjectFactory.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package com.github.cameltooling.lsp.reddeer.utils;
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/utils/JavaProjectFactory.java:1]: could you please add missing ASF license? thanks

**Review by djelinek — APPROVED** [review]

File: com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/DefaultEditorPreferencePage.java [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+		
+		// .foo and *.foo are equal in the end
+		String filetype;
+		if (type.substring(0, 1).equals(".")) {
+			filetype = "*" + type;
+		} else {
+			filetype = type;
+		}
+
+		// if file type not available, add new fil etype
```
Comment (djelinek) [inline com.github.camel-tooling.lsp.eclipse.client.tests.ui/plugins/com.github.cameltooling.lsp.reddeer/src/com/github/cameltooling/lsp/reddeer/preference/DefaultEditorPreferencePage.java:66]: typo --> "fil etype"

**Pull request merged**